lib/screens/activity_log_screen.dart

```diff
@@ -46,11 +49,94 @@ class _ActivityLogScreenState extends State<ActivityLogScreen> {
   }
 
   Widget _buildActivityEntry(BuildContext context, ActivityEntry activity) {
-    return ListTile(
-      title: Text(activity.event),
-      subtitle: Text("On this device in Work."),
-      leading: Icon(Icons.file_download_outlined),
-      trailing: Text("2021-05-26"),
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 8.0),
+      child: Column(
+        children: [
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              _getActivityName(activity),
+              _getActivityDate(activity),
+            ],
+          ),
+          const SizedBox(height: 8.0),
+          Row(
+            children: [
+              _getActivityBasedIcon(activity),
+              const SizedBox(width: 8.0),
+              _getActivityDesc(activity),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _getActivityName(ActivityEntry activity) {
+    return Text(
+      activity.event ?? '',
+      style: TextStyle(
+        fontWeight: FontWeight.bold,
+        color: Colors.black,
+        fontSize: 18,
+      ),
+    );
+  }
+
+  Widget _getActivityDate(ActivityEntry activity) {
+    return Text(
+      "${activity.timestamp?.day}/${activity.timestamp?.month}/${activity.timestamp?.year}",
+      style: TextStyle(
+        color: Colors.black.withOpacity(0.6),
+        fontSize: 14,
+      ),
+    );
+  }
+
+  Widget _getActivityBasedIcon(ActivityEntry activity) {
+    return Image.asset(
+      activity.event == "Task Completed"
+          ? "./assets/up_arrow_square.png"
+          : "./assets/download.png",
+      color: Colors.black,
+      height: 22,
+      width: 22,
+    );
+  }
+
+  Widget _getActivityDesc(ActivityEntry activity) {
+    return RichText(
+      text: TextSpan(
+        text: "On ",
```

Comment:
The description text should indicate which device initiated the change. There are two scenarios, where activities can come from. Either they are created on "this device", (A) so the currently used device, or they were synced from a different device (B).

In case of A, show: On this device in <Name of the TaskList>.
In case of B, show: On **<device name>** in <Name of the TaskList>.

Reply:
can you please explain me again .I didn't your point?

Reply:
We build a task list app. A user can create multiple task lists (currently we have only one). Each list has a name and contains tasks. Whenever a task is created or changed an activity should be logged and displayed. Task lists can be synced via P2P. When two devices sync, they exchange the tasks they have in the list. This may cause changes to tasks (e.g. "Task created", "Description Updated", "Task completed"...). These changes should be logged in activites.

The title of an activity describes what was changed (e.g. "Task created", "Description Updated", "Task completed"...).
The description of an activity should give additional context info:

- In which list did the update occur? (Name of the TaskList)
- In case of a sync, which device does the change come from (\<device name> or "this device")

The displayed activity description therefore changes depending on whether:

- The activity relates to a task that was changed on the currently used device (Scenario A)
- The activity relates to a task synced from a different device (Scenario B)